modules/kernel/src/main/scala-2.10/notebook/ReplCalculator.scala

```diff
@@ -361,7 +361,22 @@ class ReplCalculator(
       val result = scala.concurrent.Future {
         // this future is required to allow InterruptRequest messages to be received and process
         // so that spark jobs can be killed and the hand given back to the user to refine their tasks
-        val result = repl.evaluate(newCode, msg => thisSender ! StreamResponse(msg, "stdout"))
+        val cellId = er.cellId
+        def replEvaluate(code:String, cellId:String) = {
+          val cellResult = try {
+           repl.evaluate(s"""
+              |sparkContext.setJobGroup("cell-$cellId", "Created by Spark Notebook")
```

Comment:
yeah a good start. but I'd be a bit afraid of other special chars too  `"`, `\`, `\r`, `\n`.